cpp/src/arrow/ipc/CMakeLists.txt

```diff
@@ -84,13 +87,23 @@ foreach(FIL ${FBS_SRC})
   list(APPEND ABS_FBS_SRC ${ABS_FIL})
 endforeach()
 
-add_custom_command(
-  OUTPUT ${FBS_OUTPUT_FILES}
-  COMMAND ${FLATBUFFERS_COMPILER} -c -o ${OUTPUT_DIR} ${ABS_FBS_SRC}
-  DEPENDS ${ABS_FBS_SRC}
-  COMMENT "Running flatc compiler on ${ABS_FBS_SRC}"
-  VERBATIM
-)
+if(FLATBUFFERS_VENDORED)
+  add_custom_command(
+    OUTPUT ${FBS_OUTPUT_FILES}
+    COMMAND ${FLATBUFFERS_COMPILER} -c -o ${OUTPUT_DIR} ${ABS_FBS_SRC}
+    DEPENDS ${ABS_FBS_SRC} flatbuffers_ep
+    COMMENT "Running flatc compiler on ${ABS_FBS_SRC}"
+    VERBATIM
+  )
+else()
+  add_custom_command(
+    OUTPUT ${FBS_OUTPUT_FILES}
+    COMMAND ${FLATBUFFERS_COMPILER} -c -o ${OUTPUT_DIR} ${ABS_FBS_SRC}
+    DEPENDS ${ABS_FBS_SRC}
+    COMMENT "Running flatc compiler on ${ABS_FBS_SRC}"
+    VERBATIM
+  )
```

Comment:
Any issues with setting the dependencies in a cmake variable and having one `add_custom_command` call?